stitches/fx_stitch.py

```diff
@@ -0,0 +1,234 @@
+
+# Import packages
+import numpy as np
+import pandas as pd
+import xarray as xr
+import os as os
+import stitches.fx_util as util
+import stitches.fx_data as data
+import stitches.fx_pangeo as pangeo
+
+# TODO how to make sure some functions stay internal ??
+# Internal fx
+def find_zfiles(rp):
+    """ Determine which cmip files must be downlaoded from pangeo.
+        :param rp:             data frame of the recepies
+        :return:               numpy.ndarray array of the gs:// files to pull from pangeo
+    """
+
+    # Figure out which columns contain the string file
+    flat_list = rp.filter(regex='file', axis=1).values.flatten()
+    unique_list = np.unique(flat_list)
+    return unique_list
+
+# Internal fx
+def find_var_cols(x):
+    """ Determine which variables that are going to be downloaded.
+        :param x:             pandas data frame of the stitches recipe
+        :return:              a list of the variables that are going to be written out to the netcdf files.
+    """
+
+    # Parse out the variable name so that we can use it
+    # to label the final output.
+    set = x.filter(regex='file').columns.tolist()
+
+    out = []
+    for text in set:
+        new = text.replace("_file", "")
+        out.append(new)
+    return out
+
+# Internal fx
+def get_netcdf_values(i, dl, rp, fl, name):
+
+    """Extract the archive values from the list of downloaded cmip data
+        :param i:              int index of the row of the recipe data frame
+        :param dl:             list of xarray cmip files
+        :param rp:             data frame of the recipe
+        :param fl:             list of the cmip files
+        :param name:           name of the variable file that is going to be processed.
+        :return:               a slice of xarray (not sure confident on the technical term)
+    """
+
+    file = rp[name][i]
+    start_yr = str(rp["archive_start_yr"][i])
+    end_yr = str(rp["archive_end_yr"][i])
+
+    # Figure out which index level we are on and then get the
+    # xarray from the list.
+    index = int(np.where(fl == file)[0])
+    extracted = dl[index].sortby('time')
+    v=name.replace("_file", "")
+    # TODO CRV why is this function not working any more???
```

Comment:
@abigailsnyder & @crvernon the stitching function now throw an error here

```
  File "/Users/dorh012/Documents/2021/cleanup_stitch/stitches/venv/lib/python3.8/site-packages/pandas/core/indexes/base.py", line 5798, in get_slice_bound
    label = self._maybe_cast_slice_bound(label, side)
TypeError: _maybe_cast_slice_bound() missing 1 required positional argument: 'kind'
```

which is supper weird because this function has always been written in python... and worked previous so I am confused and sad :(